src/HotChocolate/Core/src/Execution/ITaskQueue.cs

```diff
@@ -1,29 +1,41 @@
+using System;
 using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
 
 namespace HotChocolate.Execution.Utilities
 {
     /// <summary>
-    /// 
+    ///  The task queue stores <see cref="ResolverTask"/> in a queue. 
     /// </summary>
     internal interface ITaskQueue
     {
         /// <summary>
-        /// 
+        /// This event is rasied when a task is enqueued
+        /// </summary>
+        event EventHandler TaskEnqueued;
+
+        /// <summary>
+        /// The number of <see cref="ResolverTask"/> in the queue
         /// </summary>
         int Count { get; }
 
         /// <summary>
-        /// 
+        /// Checks if elements are in the queue. Returns <c>true</c> when the queue has no
+        /// elements.
         /// </summary>
         bool IsEmpty { get; }
 
         /// <summary>
-        /// 
+        /// Tried to dequeue a element from the queue
         /// </summary>
-        bool TryDequeue(out ResolverTask task);
+        /// <param name="task">
+        ///     The dequed task. Is not null when the method returns<c>true</c>
```

Comment:
remove indentation

should look like this
/// <param name="task">
/// The dequed task. Is not null when the method returns<c>true</c>